Extensions/xkit_preferences.js

```diff
@@ -1080,10 +1103,20 @@ XKit.extensions.xkit_preferences = new Object({
 			obj.icon = XKit.extensions.xkit_preferences.default_extension_icon;
 		}
 
-		if (XKit.installed.check(obj.name) === true) { return ""; }
-		if (obj.name.startsWith("xkit_")) { return ""; }
+		if (obj.name.startsWith("xkit_") && XKit.tools.get_setting("xkit_show_internals","false") === "false") { return ""; }
+
+		var blacklisted_extensions = ["xkit_installer"];
+
+		for(var i = 0; i < blacklisted_extensions.length; i++) {
```

Comment:
I would suggest [`arr.indexOf()`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) instead of a `for` loop.